miscellaneous/roboconf-doc-generator/src/test/java/net/roboconf/doc/generator/RendererManagerFopTest.java

```diff
@@ -0,0 +1,99 @@
+ * as a "Result" pursuant to the terms and conditions of the LINAGORA
+ * - Université Joseph Fourier - Floralis research program. Each copyright
+ * holder of Results enumerated here above fully & independently holds complete
+ * ownership of the complete Intellectual Property rights applicable to the whole
+ * of said Results, and may freely exploit it in any manner which does not infringe
+ * the moral rights of the other copyright holders.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.roboconf.doc.generator;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.sax.SAXResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.fop.apps.Fop;
+import org.apache.fop.apps.FopFactory;
+import org.junit.Assert;
+import org.junit.Test;
+
+import net.roboconf.core.internal.tests.TestUtils;
+import net.roboconf.doc.generator.RenderingManager.Renderer;
+
+
+
+/**
+ * @author Amadou Diarra - UGA
+ */
+
+public class RendererManagerFopTest extends AbstractTestForRendererManager {
+
+
+	@Test
+	public void testFopRenderer() throws Exception {
+
+		this.rm.render( this.outputDir, this.alr.getApplicationTemplate(), this.applicationDirectory, Renderer.FOP, null );
+		File fofile = new File(this.outputDir,"index.fo");
+		Assert.assertTrue( fofile.exists() );
+		Assert.assertTrue( fofile.length() > 0 );
+
+	}
+
+	@Test
+	public void testPdfRenderer() throws Exception {
+		this.rm.render( this.outputDir, this.alr.getApplicationTemplate(), this.applicationDirectory, Renderer.FOP, null );
+		File fofile = new File(this.outputDir,"index.fo");
+		File pdffile = new File(this.outputDir,"index.pdf");
+		Assert.assertTrue( validateFop2pdf(fofile, pdffile) );
+
+	}
+
+	/**
+	 * Verifies that a FOP file is valid and transform it into pdf file.
+	 * @param fopFile the fop content
+	 * @param pdffile the output
+	 * @return true if it is valid, false otherwise
+	 * @throws Exception
+	 */
+	private boolean validateFop2pdf( File fopFile, File pdffile ) throws Exception {
+
+		File fopConfig = TestUtils.findTestFile( "/fop.xconf" );
+		OutputStream out = new BufferedOutputStream(new FileOutputStream(pdffile));
```

Comment:
If you run FindBugs on this class, you would have a warning indicating you might not close a stream.
Your code should look like...

```java
OutputStream out = null;
try {
       out = new BufferedOutputStream(new FileOutputStream(pdffile));
} finally {
       Utils.closeQuietly( out );
}
```

This way, no matter what, the output stream will always be closed.
Such error is a frequent memory leak. This is why you must care about it. :wink: